src/SDK/MonitoringApplianceEmulator/Models/SmartDetectorRunner.cs

```diff
@@ -369,5 +397,23 @@ private List<ResourceIdentifier> GetTargetResourcesForDetector(HierarchicalResou
 
             return workspaceIdToWorkspaceResourceIdMapping;
         }
+
+        /// <summary>
+        /// Gets a valid log name for running the detector on the specified emulated time
+        /// </summary>
+        /// <param name="emulatedTime">The detector's run emulated time.</param>
+        /// <returns>The log name for the run.</returns>
+        private string GetValidLogName(DateTime emulatedTime)
+        {
+            string logName = $"{emulatedTime:yyyy-MM-dd HH-mm}";
```

Comment:
>string logName = $"{emulatedTime:yyyy-MM-dd HH-mm}"; [](start = 12, length = 52)

Why not just add seconds?
Also, please add some prefix to the filename, like EmulatorRunLog (or whatever)